lib/sudo-pam/src/rpassword.rs

```diff
@@ -0,0 +1,105 @@
+/// Derived work from rpassword and rtoolbox, Copyright (c) 2023, Conrad Kleinespel et al
+///
+/// This module contains code that was originally written by Conrad Kleinespel for the rpassword
+/// crate. No copyright notices were found in the original code.
+
+/// See: https://docs.rs/rpassword/latest/rpassword/
+///
+/// CHANGES TO THE ORIGINAL CODE:
+/// - {prompt,read}_password_from_bufread deleted (we don't need them)
+/// - rtool_box::print_tty was inlined
+/// - SafeString was removed and replaced with more general Secure<> type (and moved to cutils);
+///   also, the original code actually allowed the string to quickly escape the security net
+/// - instead of String, password are read as [u8]
+/// - this also removes the need for 'fix_line_issues', since we know we read a \n
+use std::io::{self, BufRead, Write};
+use std::mem;
+use std::os::unix::io::AsRawFd;
+
+use libc::{c_int, tcsetattr, termios, ECHO, ECHONL, TCSANOW};
+
+use sudo_cutils::Secure;
+
+struct HiddenInput {
+    fd: i32,
+    term_orig: termios,
+}
+
+impl HiddenInput {
+    fn new(fd: i32) -> io::Result<HiddenInput> {
+        // Make two copies of the terminal settings. The first one will be modified
+        // and the second one will act as a backup for when we want to set the
+        // terminal back to its original state.
+        let mut term = safe_tcgetattr(fd)?;
+        let term_orig = safe_tcgetattr(fd)?;
+
+        // Hide the password. This is what makes this function useful.
+        term.c_lflag &= !ECHO;
+
+        // But don't hide the NL character when the user hits ENTER.
+        term.c_lflag |= ECHONL;
+
+        // Save the settings for now.
+        io_result(unsafe { tcsetattr(fd, TCSANOW, &term) })?;
+
+        Ok(HiddenInput { fd, term_orig })
+    }
+}
+
+impl Drop for HiddenInput {
+    fn drop(&mut self) {
+        // Set the the mode back to normal
+        unsafe {
+            tcsetattr(self.fd, TCSANOW, &self.term_orig);
+        }
+    }
+}
+
+/// Turns a C function return into an IO Result
+fn io_result(ret: c_int) -> std::io::Result<()> {
```

Comment:
We already have a function for this in our codebase, I think it is better to switch to that one instead.